supervisor/iptables_test.go

```diff
@@ -105,3 +107,103 @@ func TestSuperviseACLs(t *testing.T) {
 	}
 
 }
+
+
+//Call Supervise and we will mock AddContainer Chain here  by changing the mock defintions of newChain
+//The function should cleanup on all errors 
+func TestAddContainerChain(t *testing.T){
+	s := doNewIPTSupervisor(t)
+	containerInfo := policy.NewPUInfo("12345")
+	containerInfo.Policy.IngressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+
+	containerInfo.Policy.EgressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+	containerInfo.Runtime.SetIPAddresses(map[string]string{"bridge": "30.30.30.30"})
+	containerInfo.Policy.PolicyIPs = []string{"30.30.30.30"}
+	m := s.ipt.(TestIptablesProvider)
+
+	m.MockNewChain(t,func(table,chain string)(error){
+		if(table == "raw" ){
```

Comment:
THe same new chain function is called from multiple places in addComtaomer chain function for different tables. 
Without this check we will always fail on the first check and never excercise the other paths